pkg/apis/planetscale/v2/vitessshard_methods.go

```diff
@@ -151,3 +153,47 @@ func (s *VitessShardSpec) CellInCluster(cellName string) bool {
 	_, inZoneMap := s.ZoneMap[cellName]
 	return inZoneMap
 }
+
+// NewVitessShardCondition returns a VitessShardCondition object based on the provided type and initial state.
+func NewVitessShardCondition(ty VitessShardConditionType, initState corev1.ConditionStatus) *VitessShardCondition {
+	now := v1.NewTime(time.Now())
+	return &VitessShardCondition{
+		Type:               ty,
+		Status:             initState,
+		LastTransitionTime: &now,
+		Reason:             "initState",
+		Message:            "The initial state for this VitessShardCondition.",
+	}
+}
+
+// ChangeStatus changes the status if the current status is not the same as the new status, and updates the
+// last transition time.
+func (c *VitessShardCondition) ChangeStatus(newStatus corev1.ConditionStatus, reason, message string) {
+	// We should update reason and message regardless of whether the status type is different.
+	c.Reason = reason
+	c.Message = message
+	if c.Status == newStatus {
+		return
+	}
+
+	c.Status = newStatus
+	now := v1.NewTime(time.Now())
+	c.LastTransitionTime = &now
+}
+
+// Duration returns the duration since LastTransitionTime. It represents how long we've been in the current status for
+// this condition.
+func (c *VitessShardCondition) Duration() time.Duration {
+	return time.Now().Sub(c.LastTransitionTime.Time)
```

Comment:
There's a nice helper that makes these easier to read:

```suggestion
	return time.Since(c.LastTransitionTime.Time)
```

Reply:
Done https://github.com/planetscale/vitess-operator/pull/31/commits/7090ba557e9ebe152b564df4b9180278be40ee57.